### ISSUE TITLE
python-jwt will install in python 2 under Mac OS

### ISSUE THREAD

**Issue by zrd** [GehirnInc/python-jwt#13]
On Mac OS 10.13.5, python-jwt will install via pip to a python2 installation, seemingly successfully (see below). Installing to a python3 installation using pip3 works as expected. Installing to python2 under Ubuntu 16.04 fails at building wheel for cryptography step. Seems like a more robust version check is needed.

```
(test-venv) username@hostname:~/test-venv$ python --version
Python 2.7.10
(test-venv) username@hostname:~/test-venv$ pip install jwt
Collecting jwt
Collecting cryptography==1.7.2 (from jwt)
  Using cached https://files.pythonhosted.org/packages/51/d9/91266ca5bc54de4882d8a2e836a3038e4cb9a1cc189a9c745516dc685ea1/cryptography-1.7.2-cp27-cp27m-macosx_10_6_intel.whl
Collecting typing==3.5.3.0 (from jwt)
Collecting six>=1.4.1 (from cryptography==1.7.2->jwt)
  Using cached https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools>=11.3 in ./lib/python2.7/site-packages (from cryptography==1.7.2->jwt) (40.0.0)
Collecting pyasn1>=0.1.8 (from cryptography==1.7.2->jwt)
  Using cached https://files.pythonhosted.org/packages/a0/70/2c27740f08e477499ce19eefe05dbcae6f19fdc49e9e82ce4768be0643b9/pyasn1-0.4.3-py2.py3-none-any.whl
Collecting cffi>=1.4.1 (from cryptography==1.7.2->jwt)
  Using cached https://files.pythonhosted.org/packages/7e/4a/b647e46faaa2dcfb16069b6aad2d8509982fd63710a325b8ad7db80f18be/cffi-1.11.5-cp27-cp27m-macosx_10_6_intel.whl
Collecting enum34 (from cryptography==1.7.2->jwt)
  Using cached https://files.pythonhosted.org/packages/c5/db/e56e6b4bbac7c4a06de1c50de6fe1ef3810018ae11732a50f15f62c7d050/enum34-1.1.6-py2-none-any.whl
Collecting ipaddress (from cryptography==1.7.2->jwt)
  Using cached https://files.pythonhosted.org/packages/fc/d0/7fc3a811e011d4b388be48a0e381db8d990042df54aa4ef4599a31d39853/ipaddress-1.0.22-py2.py3-none-any.whl
Collecting idna>=2.0 (from cryptography==1.7.2->jwt)
  Using cached https://files.pythonhosted.org/packages/4b/2a/0276479a4b3caeb8a8c1af2f8e4355746a97fab05a372e4a2c6a6b876165/idna-2.7-py2.py3-none-any.whl
Collecting pycparser (from cffi>=1.4.1->cryptography==1.7.2->jwt)
Installing collected packages: six, pyasn1, pycparser, cffi, enum34, ipaddress, idna, cryptography, typing, jwt
Successfully installed cffi-1.11.5 cryptography-1.7.2 enum34-1.1.6 idna-2.7 ipaddress-1.0.22 jwt-0.5.2 pyasn1-0.4.3 pycparser-2.18 six-1.11.0 typing-3.5.3.0
(test-venv) username@hostname:~/test-venv$ python
Python 2.7.10 (default, Oct  6 2017, 22:29:07) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import jwt
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/username/test-venv/lib/python2.7/site-packages/jwt/__init__.py", line 17, in <module>
    from .jwk import (
  File "/Users/username/test-venv/lib/python2.7/site-packages/jwt/jwk.py", line 60
    def is_sign_key(self) -> bool:
                          ^
SyntaxError: invalid syntax
```

**Comment by yosida95** [comment]
@zrd Please provide your pip's version.

**Comment by zrd** [comment]
@yosida95 I'm using pip version 10.0.1, thanks.

**Comment by yosida95** [comment]
We are putting `python_requires >= 3.4` option which is supported by `pip >= 9.0.0` since jwt 0.5.3 release, so pip prevents installing `jwt >= 0.5.3` as expected but install `jwt <= 0.5.2` instead which does not have  python_requires. This is not the intended behavior, but I don't want to replace old releases with modifies.

**Comment by zrd** [comment]
So presumably jwt >= 0.5.3 will become available via `pip install jwt` in the near future? That seems a sufficient fix.

**Comment by yosida95** [comment]
Actually the latest jwt version is 0.5.4 https://pypi.org/project/jwt/0.5.4/. But it does not run on Python < 3.4 and we are not planning to support Python < 3.4. jwt is only available on Python >= 3.4.
